bentoml/_internal/frameworks/torchscript.py

```diff
@@ -103,104 +114,98 @@ def save(
-                self.linear1 = nn.Linear(context_size * embedding_dim, 128)
-                self.linear2 = nn.Linear(128, vocab_size)
+            def __init__(self, hidden_dim: int = 128, learning_rate: float = 0.0001):
+                super().__init__()
+                self.save_hyperparameters()
 
-            def forward(self, inputs):
-                embeds = self.embeddings(inputs).view((1, -1))
-                out = F.relu(self.linear1(embeds))
-                out = self.linear2(out)
-                log_probs = F.log_softmax(out, dim=1)
-                return log_probs
+                self.l1 = torch.nn.Linear(28 * 28, self.hparams.hidden_dim)
+                self.l2 = torch.nn.Linear(self.hparams.hidden_dim, 10)
 
-        _model = NGramLanguageModeler(len(vocab), EMBEDDING_DIM, CONTEXT_SIZE)
-        model = torch.jit.script(_model)
-        tag = bentoml.torchscript.save("ngrams", model)
-        # example tag: ngrams:20201012_DE43A2
+            def forward(self, x):
+                x = x.view(x.size(0), -1)
+                x = torch.relu(self.l1(x))
+                x = torch.relu(self.l2(x))
+                return x
 
-    Integration with Torch Hub and BentoML:
+            def training_step(self, batch, batch_idx):
+                x, y = batch
+                y_hat = self(x)
+                loss = F.cross_entropy(y_hat, y)
+                return loss
 
-    .. code-block:: python
+            def validation_step(self, batch, batch_idx):
+                x, y = batch
+                y_hat = self(x)
+                loss = F.cross_entropy(y_hat, y)
+                self.log("valid_loss", loss)
 
-        import torch
-        import bentoml
+            def test_step(self, batch, batch_idx):
+                x, y = batch
+                y_hat = self(x)
+                loss = F.cross_entropy(y_hat, y)
+                self.log("test_loss", loss)
 
-        resnet50 = torch.hub.load("pytorch/vision", "resnet50", pretrained=True)
-        ...
-        # trained a custom resnet50
+            def configure_optimizers(self):
+                return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)
 
-        tag = bentoml.torchscript.save("resnet50", resnet50)
-    """  # noqa
-    context: t.Dict[str, t.Any] = {
-        "framework_name": "torch",
-        "pip_dependencies": [f"torch=={get_pkg_version('torch')}"],
-    }
+        tag = bentoml.pytorch_lightning.save("lit_classifier", LitClassifier())
+    """
+
+    if _include_pytorch_lightning_version:
+        framework_versions = {
+            "torch": get_pkg_version("torch"),
+            "pytorch_lightning": get_pkg_version("pytorch_lightning"),
+        }
+    else:
+        framework_versions = {"torch": get_pkg_version("torch")}
+
+    context: ModelContext = ModelContext(
+        framework_name="torchscript",
+        framework_versions=framework_versions,
+    )
+
+    if signatures is None:
```

Comment:
does it make sense also to apply a default signature for all PyTorch models, with { "__call__": { batchable: True }}` ?

Reply:
Should we be enabling `batchable` by default?

Reply:
@sauyon I think it make sense for frameworks like PyTorch and TF, although I understand it might be better to keep the default behavior consistent and ask users to set this explicitly.

Btw we currently have batchable=True as default in the Keras module

Reply:
@parano @sauyon Pytorch modules are not using `__call__` as the default inference method, nor in its official guide, thus it doesn't make sense, I believe.

Reply:
Let me check about the original usage or __call__ in its doc

Reply:
Just updated. will use __call__ as the default method. But I don't think we should have batchable=True for models, which may cause unwanted batching behavior